ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/tls/TlsConfiguration.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.api.tls;
+
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Optional;
+
+import io.vertx.core.net.PfxOptions;
+import io.vertx.core.net.TrustOptions;
+import org.apache.tuweni.net.tls.VertxTrustOptions;
+
+public class TlsConfiguration {
+  private final Path keyStorePath;
+  private final String keyStorePassword;
+  private final Path knownClientsFile;
+  private TrustOptions trustOptions;
+  private PfxOptions pfxKeyCertOptions;
+
+  private TlsConfiguration(
+      final Path keyStorePath, final String keyStorePassword, final Path knownClientsFile) {
+    Objects.requireNonNull(keyStorePath, "Key Store Path must not be null");
```

Comment:
In other parts of Besu we've used Preconditions.checkNotNull - should this be moved to Preconditions, or is Objects the new version of Preconditions?

Reply:
I'll move the checks to Preconditions.checkNotNull. Objects.requireNonNull is JDK's built-in and it raises a nullpointerexception instead of illegalstateexception.

Reply:
I just checked it, Guava's checkNotNull is doing exact same as thing java.util.Objects.requireNotNull (JDK8+). I would favour in removing checkNotNull to use requireNotNull instead.